lib/vdsm/network/link/setup.py

```diff
@@ -20,38 +20,64 @@
 from __future__ import absolute_import
 from __future__ import division
 
-from vdsm.network.link.bridge import Bridge
+
+class NmstateBridgeOptionNotSupported(Exception):
+    pass
 
 
 def parse_bond_options(options):
     """
     Parse bonding options into a dictionary.
     """
-
-    def _string_to_dict(str, div, eq):
-        if options == '':
-            return {}
-        return dict(
-            option.split(eq, 1) for option in str.strip(div).split(div)
-        )
-
     if options:
         d_options = _string_to_dict(options, ' ', '=')
         return d_options
     else:
         return {}
 
 
-def setup_custom_bridge_opts(nets):
-    for name, opts in parse_nets_bridge_opts(nets):
-        Bridge(name, opts)
+def parse_nets_bridge_opts(opts):
 
+    nmstate_options_types = {
+        'mac-ageing-time': int,
+        'group-forward-mask': int,
+        'hash-max': int,
+        'multicast-snooping': _str_to_bool,
+        'multicast-router': int,
+        'multicast-last-member-count': int,
+        'multicast-last-member-interval': int,
+        'multicast-membership-interval': int,
+        'multicast-querier': _str_to_bool,
+        'multicast-querier-interval': int,
+        'multicast-query-use-ifaddr': _str_to_bool,
+        'multicast-query-interval': int,
+        'multicast-query-response-interval': int,
+        'multicast-startup-query-count': int,
+        'multicast-startup-query-interval': int,
+    }
 
-def parse_nets_bridge_opts(nets):
-    for name, opts in nets.items():
-        opts_str = opts.get('custom', {}).get('bridge_opts')
-        if opts_str:
-            bridge_opts = dict(
-                opt.split('=', 1) for opt in opts_str.split(' ')
-            )
-            yield name, bridge_opts
+    if not opts:
+        return {}
+    opts_str = opts.replace('_', '-')
+    bridge_opts = _string_to_dict(opts_str, ' ', '=')
+    try:
+        for key in bridge_opts.keys():
+            bridge_opts[key] = nmstate_options_types[key](bridge_opts[key])
+    except KeyError:
+        raise NmstateBridgeOptionNotSupported(
+            f'{key} is not a valid nmstate birdge option'
+        )
+    return bridge_opts
+
+
+def _str_to_bool(str_value):
+    false_values = ['false', 'False', '0']
```

Comment:
We could save one value in the array by using `str_value.lower()`, but not that important.